reps/2024-06-16-support-apache-yunikorn-scheduler.md

```diff
@@ -0,0 +1,157 @@
+Option syntax: `--batch-scheduler-name [SupportedSchedulerName]`, for example:
+
+Option Syntax:
+
+```shell
+# 1. use volcano
+--enable-batch-scheduler=true
+--batch-scheduler-name=volcano
+
+# 2. use yunikorn
+--enable-batch-scheduler=true
+--batch-scheduler-name=yunikorn
+
+# 3. use default Kubernetes scheduler, or do not turn on the feature flag
+--enable-batch-scheduler=true
+--batch-scheduler-name=default
+```
+
+When a scheduler name is specified, the scheduler manager will only load the configured scheduler plugin,
+ensuring only necessary resources are loaded in the controller runtime.
+
+The option `--batch-scheduler-name` accepts a single scheduler name as the value, and the value must be `default`,
+`volcano` and `yunikorn` (before other scheduler plugins introduced). If an unrecognized scheduler name is provided,
+the controller will fail to start with an error indicating that the scheduler plugin is not found.
+
+### Deprecate "ray.io/scheduler-name"
+
+With the scheduler name is specified in the operator startup options, there is no need to set `ray.io/scheduler-name`
+in `RayJob` or `RayCluster` CRs. This option should be marked as deprecated and eventually removed.
+
+### Compatibility
+
+When this change is introduced, existing users who use `--enable-batch-scheduler` with the Volcano scheduler
+must explicitly add `--batch-scheduler-name=volcano`. Otherwise, the controller will default to the Kubernetes scheduler.
+This is an incompatible change and must be carefully documented to mitigate potential impact.
+
+### YuniKorn scheduler plugin behavior
+
+The YuniKorn scheduler plugin will support both `RayJob` and `RayCluster` resources. The integration will be lightweight,
+as YuniKorn does not require new CRDs. The plugin will add labels to the Ray pods, and the YuniKorn scheduler will
+schedule Ray pods based on these labels.
+
+To enable yunikorn scheduler, set the following options when starting the KubeRay operator:
+
+```shell
+--enable-batch-scheduler=true
+--batch-scheduler-name=yunikorn
+```
+
+if submitting a `RayCluster`, add `yunikorn.apache.org/queue-name` and `yunikorn.apache.org/application-id` to the labels.
+
+```yaml
+apiVersion: ray.io/v1
+kind: RayCluster
+metadata:
+  labels:
+    ray.io/scheduler-name: yunikorn
+    yunikorn.apache.org/queue-name: root.abc
+    yunikorn.apache.org/application-id: rayjob-sample-ltpjh 
+```
+
+The `RayJob` will be submitted to `root.abc` queue and scheduled by the yunikorn scheduler. The `RayJob` will be
+recognized as a "job" with ID "rayjob-sample-ltpjh".
+
+If submitting a `RayJob`, provide only the scheduler name and queue name:
+
+```yaml
+apiVersion: ray.io/v1
+kind: RayJob
+metadata:
+  name: rayjob-example
+  namespace: my-namespace
+  labels:
+    ray.io/scheduler-name: yunikorn
+    yunikorn.apache.org/queue-name: root.abc
+```
+
+when the Ray job is submitted to the cluster, the Ray operator will create the following RayJob CR:
```

Comment:
This says RayJob CR but below is a RayCluster example, can you update please?

Reply:
Fixed.